crates/db_views/src/comment_view.rs

```diff
@@ -53,13 +33,14 @@ fn queries<'a>() -> Queries<
   );
 
   let is_saved = |person_id| {
-    exists(
-      comment_saved::table.filter(
+    comment_saved::table
```

Comment:
@dullbananas I could use your expertise here.

So when using `options.saved_only` (IE viewing your own profiles saved posts / comments), we need to sort not by the `comment_aggregates.published`, but by `comment_saved.publish` .

This is difficult because it needs a conditional join. I can only think of three ways to do this:

1. Try to select multiple columns, like I'm doing here. I can't get this to work correctly, because the boxable expression types give a lot of errors.
2. Alter this join to only select the published time, rather than multiple values.
3. Make the join to comment_saved mandatory, then I'll be able to use `order_by(comment_saved::published)` below.

Which one do you think would be best?

Reply:
It should only select the published time. Then is_saved can be changed to saved_time, and is_saved_selection can use is_not_null.

Also, order_by should work fine with a nullable value.

Reply:
Thx! I'll give that a shot.